# Step 0 - Package Import

I'm testing Positron out for this project and I've just reinstalled R 4.4.2. So that means time to manage packages.

In [5]:
list_of_packages = c("httr","tidyverse","jsonlite","glue","janitor","naniar","ggplot2")
install.packages(list_of_packages)


Warning in install.packages(list_of_packages) :
  'lib = "C:/Program Files/R/R-4.4.2/library"' is not writable


also installing the dependencies ‘fastmap’, ‘bit’, ‘ps’, ‘base64enc’, ‘sass’, ‘digest’, ‘sys’, ‘cachem’, ‘rappdirs’, ‘rematch’, ‘bit64’, ‘prettyunits’, ‘processx’, ‘evaluate’, ‘highr’, ‘xfun’, ‘yaml’, ‘bslib’, ‘fontawesome’, ‘htmltools’, ‘jquerylib’, ‘tinytex’, ‘Rcpp’, ‘colorspace’, ‘askpass’, ‘backports’, ‘generics’, ‘memoise’, ‘blob’, ‘DBI’, ‘data.table’, ‘gargle’, ‘uuid’, ‘cellranger’, ‘ids’, ‘rematch2’, ‘cpp11’, ‘pkgconfig’, ‘timechange’, ‘utf8’, ‘systemfonts’, ‘textshaping’, ‘clipr’, ‘crayon’, ‘vroom’, ‘tzdb’, ‘progress’, ‘callr’, ‘fs’, ‘knitr’, ‘rmarkdown’, ‘selectr’, ‘fansi’, ‘viridisLite’, ‘gridExtra’, ‘plyr’, ‘farver’, ‘labeling’, ‘munsell’, ‘RColorBrewer’, ‘curl’, ‘mime’, ‘openssl’, ‘R6’, ‘broom’, ‘conflicted’, ‘cli’, ‘dbplyr’, ‘dplyr’, ‘dtplyr’, ‘forcats’, ‘googledrive’, ‘googlesheets4’, ‘haven’, ‘hms’, ‘lubridate’, ‘magrittr’, ‘modelr’, ‘pillar’, ‘purrr’, ‘ragg’, ‘readr’, ‘readxl’, ‘reprex’, ‘rlang’, ‘rstudioapi’, ‘rvest’, ‘stringr’, ‘tibble’, ‘tidyr’, ‘xml2’, ‘lifecycle’, 

trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.4/fastmap_1.2.0.zip'
Content type 'application/zip' length 135571 bytes (132 KB)
downloaded 132 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.4/bit_4.5.0.1.zip'
Content type 'application/zip' length 1182828 bytes (1.1 MB)
downloaded 1.1 MB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.4/ps_1.8.1.zip'
Content type 'application/zip' length 645290 bytes (630 KB)
downloaded 630 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.4/base64enc_0.1-3.zip'
Content type 'application/zip' length 33130 bytes (32 KB)
downloaded 32 KB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.4/sass_0.4.9.zip'
Content type 'application/zip' length 2614085 bytes (2.5 MB)
downloaded 2.5 MB

trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.4/digest_0.6.37.zip'
Content type 'application/zip' length 223481 bytes (218 KB)
downloaded 218 KB

trying URL 'https://cran.rstudio.com/bin/windows/contri

package ‘fastmap’ successfully unpacked and MD5 sums checked
package ‘bit’ successfully unpacked and MD5 sums checked
package ‘ps’ successfully unpacked and MD5 sums checked
package ‘base64enc’ successfully unpacked and MD5 sums checked
package ‘sass’ successfully unpacked and MD5 sums checked
package ‘digest’ successfully unpacked and MD5 sums checked
package ‘sys’ successfully unpacked and MD5 sums checked
package ‘cachem’ successfully unpacked and MD5 sums checked
package ‘rappdirs’ successfully unpacked and MD5 sums checked
package ‘rematch’ successfully unpacked and MD5 sums checked
package ‘bit64’ successfully unpacked and MD5 sums checked
package ‘prettyunits’ successfully unpacked and MD5 sums checked
package ‘processx’ successfully unpacked and MD5 sums checked
package ‘evaluate’ successfully unpacked and MD5 sums checked
package ‘highr’ successfully unpacked and MD5 sums checked
package ‘xfun’ successfully unpacked and MD5 sums checked
package ‘yaml’ successfully unpacked and

In [7]:
library(httr)
library(tidyverse)
library(jsonlite)
library(glue)
library(janitor)
library(naniar)
library(readr)

# Step 1 - Data Import
For this we're going to pull the cleaned data from github.

In [8]:
parfumo_data_clean <- readr::read_csv('https://raw.githubusercontent.com/rfordatascience/tidytuesday/main/data/2024/2024-12-10/parfumo_data_clean.csv')

Rows: 59325 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (10): Number, Name, Brand, Concentration, Main_Accords, Top_Notes, Middl...
dbl  (3): Release_Year, Rating_Value, Rating_Count

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Luckily, this data came with a data dictionary as seen below.
### Data Dictionary

# `parfumo_data_clean.csv`

|variable      |class       |description                                                                      |
|:-------------|:-----------|:--------------------------------------------------------------------------------|
|Number        |character   |A unique identifier or number assigned to each perfume.                          |
|Name          |character   |The name of the perfume or fragrance.                                            |
|Brand         |character   |The brand or manufacturer of the fragrance.                                      |
|Release_Year  |double      |The year the fragrance was released.                                             |
|Concentration |character   |The concentration of the fragrance (e.g., Eau de Parfum, Eau de Toilette).       |
|Rating_Value  |double      |The overall rating score given by users.                                         |
|Rating_Count  |double      |The number of user ratings for the fragrance.                                    |
|Main_Accords  |character   |The primary scent characteristics or accords of the fragrance.                   |
|Top_Notes     |character   |The initial scent notes perceived after application.                             |
|Middle_Notes  |character   |The heart or middle notes of the fragrance that emerge after the top notes fade. |
|Base_Notes    |character   |The lasting, final scent notes that linger after the fragrance has dried down.   |
|Perfumers     |character   |The creators or perfumers responsible for the fragrance composition.             |
|URL           |character   |The link to the product page on Parfumo.com.                                     |

Now that we have our data uploaded, we can start some EDA.
First, lets get an idea of the ratings look like for this dataset as a whole.

In [ ]:
parfumo_data_ratings <- parfumo_data_clean %>%
	filter(!is.na(Rating_Value)) %>%
	filter(!is.na(Rating_Count))

glimpse(parfumo_data_ratings)

While we have 59325 perfumes in this dataset, only 30,046 of them have a value for average rating and the number of ratings associated with that perfume.
Lets see what the distribution looks like of perfume ratings looks like.

perfume_ratings_histo <- parfumo_data_clean